generate_polices.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+    "fmt"
+    "bytes"
+)
+
+type ruleOption struct {
+    occurance int
+    g         generator
+}
+
+func generateAuthorizationPolicy(action string, ruleToOccurences map[string]*ruleOption) (string, error) {
+    yaml := bytes.Buffer{}
+    yaml.WriteString("spec:\n")
+    yaml.WriteString(" action: " + action + "\n")
+
+    if (len(ruleToOccurences) == 0) {
+        yaml.WriteString(" {}")
+    } else {
+        yaml.WriteString(" rules:\n")
+    }
+
+    for rule, ruleOp := range ruleToOccurences {
+        yaml.WriteString("  " + rule + ":\n")
+        rules, err := ruleOp.g.generate(rule, ruleOp.occurance)
+        if err != nil {
+            return "", err
+        }
+        yaml.WriteString(rules)
+    }
+    return yaml.String(), nil
+}
+
+func generateRule(policy string, action string, ruleToOccurences map[string]*ruleOption) (string, error) {
+ 
+    switch policy {
+    case "AuthorizationPolicy":
+        return generateAuthorizationPolicy(action, ruleToOccurences)
+    case "PeerAuthentication":
+        fmt.Println("PeerAuthentication")
+        // TODO impliment
+        // return generatePeerAuthentication(selector, mtl []*mode, portLevel)
+    case "RequestAuthentication":
+        fmt.Println("RequestAuthentication")
+        // TODO impliment
+        // return generateRequestAuthentication(selector, ruleToOccurences)
+    default:
+        fmt.Println("invalid policy")
+    }
+
+    return "", fmt.Errorf("invalid policy")
+}
+
+
+func createRules(policy string, action string, ruleToOccurences map[string]*ruleOption) (string, error){
+    yaml, err := generateRule(policy, action, ruleToOccurences)
+    if (err != nil) {
```

Comment:
no parentheses needed here. You can also do like,
`if yaml, err := generateRule(policy, action, ruleToOccurences); err != nil {}`